apps/server/src/modules/course/repo/course.repo.ts

```diff
@@ -93,4 +94,18 @@ export class CourseRepo extends BaseRepo<CourseEntity> {
 
 		return course;
 	}
+
+	public async removeUserReference(userId: EntityId): Promise<number> {
+		const id = new ObjectId(userId);
+		const count = await this._em.nativeUpdate(
+			CourseEntity,
+			{
+				$or: [{ teachers: id }, { substitutionTeachers: id }, { students: id }],
```

Comment:
Maybe outsourse querys to QueryBuilder class? Seperation of orchestration vs logic?

Reply:
was discussed above